component/src/main/java/io/siddhi/extension/io/file/listeners/FileCronExecutor.java

```diff
@@ -0,0 +1,185 @@
+/*
+ * Copyright (c) 2020, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ * WSO2 Inc. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package io.siddhi.extension.io.file.listeners;
+
+import io.siddhi.core.config.SiddhiAppContext;
+import io.siddhi.core.stream.input.source.SourceEventListener;
+import io.siddhi.extension.io.file.processors.FileProcessor;
+import io.siddhi.extension.io.file.util.Constants;
+import io.siddhi.extension.io.file.util.FileSourceConfiguration;
+import io.siddhi.extension.io.file.util.VFSClientConnectorCallback;
+import org.apache.log4j.Logger;
+import org.quartz.CronScheduleBuilder;
+import org.quartz.Job;
+import org.quartz.JobBuilder;
+import org.quartz.JobDataMap;
+import org.quartz.JobDetail;
+import org.quartz.JobExecutionContext;
+import org.quartz.JobExecutionException;
+import org.quartz.JobKey;
+import org.quartz.Scheduler;
+import org.quartz.SchedulerException;
+import org.quartz.Trigger;
+import org.quartz.TriggerBuilder;
+import org.quartz.impl.StdSchedulerFactory;
+import org.wso2.carbon.messaging.BinaryCarbonMessage;
+import org.wso2.carbon.messaging.exceptions.ClientConnectorException;
+import org.wso2.transport.file.connector.sender.VFSClientConnector;
+
+import java.io.File;
+import java.nio.ByteBuffer;
+import java.nio.charset.StandardCharsets;
+import java.util.Map;
+
+import static io.siddhi.extension.io.file.util.GenerateAppProperties.constructPath;
+import static io.siddhi.extension.io.file.util.GenerateAppProperties.generateProperties;
+import static io.siddhi.extension.io.file.util.GenerateAppProperties.getFileName;
+import static io.siddhi.extension.io.file.util.GenerateAppProperties.reProcessFileGenerateProperties;
+
+/**
+ * FileCronExecutor is executed when the cron expression is given. If the current time satisfied by the cron
+ * expression then the file processing will be executed.
+ */
+public class FileCronExecutor implements Job {
+    private static final Logger log = Logger.getLogger(FileCronExecutor.class);
+
+    public FileCronExecutor() {
+    }
+
+    public static void scheduleJob(FileSourceConfiguration fileSourceConfiguration,
```

Comment:
Shall we add Javadoc comments for all the new methods?

Reply:
fixed in a6f1f669af7b573c96bc9d004d0244e4dc9c7e5c